src/export/StructureDefinitionExporter.ts

```diff
@@ -201,6 +217,43 @@ export class StructureDefinitionExporter implements Fishable {
     }
   }
 
+  applyDeferredRules() {
+    this.deferredRules.forEach((rules, sd) => {
+      for (const rule of rules) {
+        if (typeof rule.value === 'string') {
+          const fishedValue = this.fishForFHIR(rule.value);
+          // an inline instance will fish up an InstanceDefinition, which can be used directly.
+          // other instances will fish up an Object, which needs to be turned into an InstanceDefinition.
+          // an InstanceDefinition of a resource needs a resourceType, so check for that property.
+          // if we can't find a resourceType or an sdType, we have a non-resource Instance, which is no good
+          if (fishedValue) {
+            if (
+              fishedValue instanceof InstanceDefinition &&
+              (fishedValue.resourceType || fishedValue._instanceMeta?.sdType)
```

Comment:
I'm not sure if the `fishedValue._instanceMeta?.sdType` comes into play here. As far as I can tell, `this.fishForFHIR` calls `fisher.fishForFHIR`, which in turn calls `pkg.fishForFHIR`, which strips instanceMeta on an instance by calling `toJSON` I believe. At least that is how I am tracing it, but it's a confusing thing to track down so I may be missing something.

Reply:
Ah, that's a good point. I was trying to make sure it would be ready for `ElementDefinition.checkFixInlineInstance`, but when I removed the check for `sdType`, none of the tests broke. So either my tests have missed something, or that check isn't necessary.